doc/source/file_formats.rst

```diff
@@ -1,23 +1,375 @@
+parameters. For all interactions, the parameters are read as is and
+written to the output topology without interpreting and/or
+checking. Bond/constraint lengths, angles and dihedral angles may be
+used for generating missing coordinates.
+
+A full list of interactions is given below, corresponding to the list
+of intramolecular interactions available in Gromacs, with a number
+specifying the number of particles involved in the interaction. Note
+that improper dihedrals are listed as a separate interaction type,
+whereas in Gromacs these fall under the dihedrals section.
+
+* bonds(2)
+* angles(3)
+* dihedrals(4)
+* impropers(4)
+* constraints(2)
+* pairs(2)
+* pairs_nb(2)
+* SETTLE(1)
+* virtual_sites2(3)
+* virtual_sites3(4)
+* virtual_sites4(5)
+* position_restraints(1)
+* distance_restraints(2)
+* dihedral_restraints(4)
+* orientation_restraints(2)
+* angle_restraints(4)
+* angle_restraints_z(2)
+* cmap(...)
+
+Any of the subsections can be given multiply times, in which case they
+are additive. Do note that in the output topology specifying the same
+interaction several times (the same type and particles) will overrule
+any previous one, except when they are given different contexts (see
+below).
+
+Links
+~~~~~
+
+To generate a topology for a polymer or any molecule constructed from
+joining parts, Vermouth connects moleculetypes using links. A link
+describes how blocks are to be joined, what changes are effected in
+the atom lists and which interactions are added, removed, or
+altered. The changes in the atom and interaction lists are specified
+using the corresponding subsections as under moleculetype. Further
+subsections available for links are patterns, features, molmeta,
+edges, and non-edges
+
+The lines following the section tag may list selection statements for
+filtering atoms in which to search for matching patterns. Each line
+specifies a property and the corresponding value. The selection
+statements may include filters based on, e.g., the residue name and
+the secondary structure type, which are used to determine the
+structural properties of protein backbone in the Martini force field.
+
+Note: the link should have a name as overall content, to be consistent
+with moleculetype and modification (and to increase clarity), and the
+filters should probably fall under a subsection [ filters ].
+
+Patterns
+^^^^^^^^
+
+Patterns is a subsection that lists patterns of atoms to which the
+link applies. Each line in the subsection describes a pattern. One of
+the patterns must apply for the link to match. A pattern consists of
+atom identifiers. Each atom identifier consists of a name which may be
+preceded by a shift and which may be further specified using an
+attributes statement. The shift indicates a relative position in terms
+of residues:
+
+* +, ++, +++ : 	first, second, third following residue
+* -, --, --- : 	first, second, third previous residue
+* > :		other residue (not in order)
```

Comment:
`>` means residue with larger resid (smaller also exists).
`*` means residue with different resid (smaller or larger). Beware of symmetry hell. But there's an open PR about this #402